benchmark/training/training_benchmark.py

```diff
@@ -0,0 +1,189 @@
+import argparse
+import ast
+
+import torch
+import torch.nn.functional as F
+from tqdm import tqdm
+from utils import get_dataset, get_model
+
+from torch_geometric.loader import NeighborLoader
+from torch_geometric.nn import PNAConv
+from torch_geometric.profile import rename_profile_file, timeit, torch_profile
+
+supported_sets = {
+    'ogbn-mag': ['rgat', 'rgcn'],
+    'ogbn-products': ['edge_cnn', 'gat', 'gcn', 'pna', 'sage'],
+    'Reddit': ['edge_cnn', 'gat', 'gcn', 'pna', 'sage'],
+}
+
+
+def train(model, loader, optimizer, device, hetero_flow, progress_bar=True,
+          desc="") -> None:
+    if progress_bar:
+        loader = tqdm(loader, desc=desc)
+    for batch in loader:
+        optimizer.zero_grad()
+        batch = batch.to(device)
+        if hetero_flow:
```

Comment:
As checking this condition each time can be time-consuming, I think we can move this "if" outside of the loop iterating over the batches and have two separate loops instead. Or have two train functions - one for hetero and second for homo graphs. What do You think?